djangocms_navigation/admin.py

```diff
@@ -59,10 +71,170 @@ class MenuContentAdmin(admin.ModelAdmin):
     form = MenuContentForm
     menu_model = Menu
     menu_item_model = MenuItem
-    list_display = ["title", "get_menuitem_link", "get_preview_link"]
     list_display_links = None
     list_filter = (LanguageFilter, )
 
+    class Media:
+        css = {"all": ("djangocms_versioning/css/actions.css", "djangocms_version_locking/css/version-locking.css",)}
+
+    def get_version(self, obj):
+        """
+        Return the latest version of a given object
+        :param obj: MenuContent instance
+        :return: Latest Version linked with MenuContent instance
+        """
+        return obj.versions.all()[0]
+
+    def get_state_display(self, obj):
+        """
+        Return the current state of the version
+        :param obj: MenuContent Instance
+        :return: State
+        """
+        return self.get_version(obj).get_state_display()
+
+    get_state_display.short_description = _("State")
+
+    def get_author(self, obj):
+        """
+        Return the author who created a version
+        :param obj: MenuContent Instance
+        :return: Author
+        """
+        return self.get_version(obj).created_by
+
+    get_author.short_description = _("Author")
+
+    def get_modified_date(self, obj):
+        """
+        Get the last modified date of a version
+        :param obj: MenuContent Instance
+        :return: Modified Date
+        """
+        return self.get_version(obj).modified
+
+    get_modified_date.short_description = _("Modified")
+
+    def _list_actions(self, request):
+        """
+        A closure that makes it possible to pass request object to
+        list action button functions.
+        """
+
+        def list_actions(obj):
+            """Display links to state change endpoints
+            """
+            return format_html_join(
+                "",
+                "{}",
+                ((action(obj, request),) for action in self.get_list_actions()),
+            )
+
+        list_actions.short_description = _("actions")
+        return list_actions
+
+    def get_list_actions(self):
+        return [
+            self._get_preview_link,
+            self._get_edit_link,
+        ]
+
+    def get_list_display(self, request):
+        menu_content_list_display = ["title"]
+        versioning_enabled = is_versioning_enabled(self.model)
+
+        if versioning_enabled:
+            menu_content_list_display.extend(
+                ["get_author", "get_modified_date"]
+            )
+            # Add version locking specific items
+            if using_version_lock:
+                menu_content_list_display.extend(["get_state_display"])
+                # Ensure actions are the last items
+                menu_content_list_display.extend([self._list_actions(request)])
+            else:
+                menu_content_list_display.extend(["get_menuitem_link", "get_preview_link"])
```

Comment:
I don't think that these items should be blocked by version locking. If you look at the proposed solution you'll see that:

The following are shown when versioning is not present // enabled: 
    menu_content_list_display.extend(["get_menuitem_link", "get_preview_link"])

The following are shown when versioning is present:
    menu_content_list_display.extend(["get_versioning_state", "get_author", "get_modified_date", "get_state_display"]
    menu_content_list_display.extend([self._list_actions(request)])


The following is shown when version locking is enabled:
        menu_content_list_display.extend(["is_locked"])


The modified version above does not respect the rules listed here.

Reply:
Also is locked has now disappeared altogether. Please simply copy and paste what I proposed before with the `get_versioning_state` entry removed.

Reply:
This logic is still wrong and doesn't represent the previous comments. If I removed locking I woudl also remove things that are for versioning which is not right.